tests/functional/RemoteWebElementTest.php

```diff
@@ -117,6 +118,65 @@ public function testShouldClick()
         );
     }
 
+    /**
+     * Test that workarounds to geckodriver#653 works as expected.
+     *
+     * @covers ::click
+     * @covers ::clickChildElement
+     * @dataProvider provideValidClick
+     * @param string $id
+     */
+    public function testShouldClickOnBlockLevelElement($id)
+    {
+        $this->driver->get($this->getTestPageUrl('links.html'));
+        $linkElement = $this->driver->findElement(WebDriverBy::id($id));
+
+        $linkElement->click();
+        $this->assertStringContainsString('index.html', $this->driver->getCurrentUrl());
+    }
+
+    /**
+     * @return array
+     */
+    public function provideValidClick()
+    {
+        return [
+            'Standard text link' => ['a-index-plain'],
+            'First child as block-level element' => ['a-index-block-child'],
+            'First block-level child hidden' => ['a-index-block-child-hidden'],
+            'Second block-level child hidden' => ['a-index-second-child-hidden'],
+        ];
+    }
+
+    /**
+     * Test that workarounds to geckodriver#653 do not prevent hidden elements from throwing an
+     * ElementNotInteractableException.
+     *
+     * @covers ::click
+     * @covers ::clickChildElement
+     * @dataProvider provideInvalidClick
+     * @param string $id
+     */
+    public function testShouldClickNotInteractable($id)
+    {
+        $this->driver->get($this->getTestPageUrl('links.html'));
+        $linkElement = $this->driver->findElement(WebDriverBy::id($id));
+
+        $this->expectException(ElementNotInteractableException::class);
+        $linkElement->click();
+    }
+
+    /**
+     * @return array
+     */
+    public function provideInvalidClick()
```

Comment:
Dtto, let's make it single test?

Reply:
This one is more of a pain because of the try/catch but there's only two test conditions so easy enough.